pkg/apiserver/registry/security/workloadidentity/storage/jwt.go

```diff
@@ -90,65 +93,139 @@ func (r *TokenRequestREST) getGardenerClaims(workloadIdentity *securityapi.Workl
-			return nil, nil, nil, err
+		if seed, err = r.seedLister.Get(ctxObj.Name); err != nil {
+			return nil, nil, nil, nil, nil, err
 		}
 
 		if seed.UID != ctxObj.UID {
-			return nil, nil, nil, fmt.Errorf("uid of contextObject (%s) and real world resource(%s) differ", ctxObj.UID, seed.UID)
+			return nil, nil, nil, nil, nil, fmt.Errorf("uid of contextObject (%s) and real world resource(%s) differ", ctxObj.UID, seed.UID)
 		}
 		seedMeta = seed.GetObjectMeta()
 
+	case gvk.Group == gardencorev1beta1.SchemeGroupVersion.Group && gvk.Kind == "BackupBucket":
+		if backupBucket, err = r.backupBucketLister.Get(ctxObj.Name); err != nil {
+			return nil, nil, nil, nil, nil, err
+		}
+
+		if backupBucket.UID != ctxObj.UID {
+			return nil, nil, nil, nil, nil, fmt.Errorf("uid of contextObject (%s) and real world resource(%s) differ", ctxObj.UID, backupBucket.UID)
+		}
+		backupBucketMeta = backupBucket.GetObjectMeta()
+
+		if backupBucket.Spec.SeedName != nil {
+			seedName := *backupBucket.Spec.SeedName
+			if seed, err = r.seedLister.Get(seedName); err != nil {
+				return nil, nil, nil, nil, nil, err
+			}
+			seedMeta = seed.GetObjectMeta()
+		}
+
+	case gvk.Group == gardencorev1beta1.SchemeGroupVersion.Group && gvk.Kind == "BackupEntry":
+		if backupEntry, err = r.backupEntryLister.BackupEntries(*ctxObj.Namespace).Get(ctxObj.Name); err != nil {
+			return nil, nil, nil, nil, nil, err
+		}
+
+		if backupEntry.UID != ctxObj.UID {
+			return nil, nil, nil, nil, nil, fmt.Errorf("uid of contextObject (%s) and real world resource(%s) differ", ctxObj.UID, backupEntry.UID)
+		}
+		backupEntryMeta = backupEntry.GetObjectMeta()
+
+		if backupBucket, err = r.backupBucketLister.Get(backupEntry.Spec.BucketName); err != nil {
+			return nil, nil, nil, nil, nil, err
+		}
+		backupBucketMeta = backupBucket.GetObjectMeta()
+
+		if backupEntry.Spec.SeedName != nil {
+			seedName := *backupEntry.Spec.SeedName
+			if seed, err = r.seedLister.Get(seedName); err != nil {
+				return nil, nil, nil, nil, nil, err
+			}
+			seedMeta = seed.GetObjectMeta()
+		}
+
+		if shootName := gardenerutils.GetShootNameFromOwnerReferences(backupEntry); shootName != "" {
+			shoot, err := r.shootListers.Shoots(backupEntry.GetNamespace()).Get(shootName)
+			if err != nil {
+				return nil, nil, nil, nil, nil, err
+			}
+
+			shootTechnicalID, shootUID := gardenerutils.ExtractShootDetailsFromBackupEntryName(ctxObj.Name)
+			if shootTechnicalID == shoot.Status.TechnicalID && shootUID == shoot.GetUID() {
+				shootMeta = shoot.GetObjectMeta()
+
+				if project, err = admissionutils.ProjectForNamespaceFromLister(r.projectLister, shoot.Namespace); err != nil {
+					return nil, nil, nil, nil, nil, err
+				}
+				projectMeta = project.GetObjectMeta()
+			}
+		}
+
 	default:
-		return nil, nil, nil, fmt.Errorf("unsupported GVK for context object: %s", gvk.String())
+		return nil, nil, nil, nil, nil, fmt.Errorf("unsupported GVK for context object: %s", gvk.String())
```

Comment:
I suggest that we group these in a single struct as it will be easier to read.

Reply:
Sure, I will do it.

Reply:
Done with 14a023f22dadaf481bfa69cab155f7c7377e786a